frontend/app/common/fetcher.ts

```diff
@@ -13,38 +14,67 @@ export const XSRF_HEADER = 'X-XSRF-TOKEN';
 /** Cookie field with XSRF token */
 export const XSRF_COOKIE = 'XSRF-TOKEN';
 
-type Methods = typeof METHODS;
-type FetchInit = Omit<RequestInit, 'headers'> & {
-  headers?: Record<string, string>;
-  json?: unknown;
-  query?: Record<string, string | number | undefined>;
-};
-type FetcherObject = Record<Methods[number], <T>(url: string, params?: FetchInit) => Promise<T>>;
+type QueryParams = Record<string, string | number | undefined>;
+type Payload = BodyInit | Record<string, unknown> | null;
+type FetcherMethods = Record<'get' | 'delete', <T>(url: string, query?: QueryParams) => Promise<T>> &
+  Record<'put' | 'post', <T>(url: string, query?: QueryParams, body?: Payload) => Promise<T>>;
+
+export function stringifyUrl(uri: string, query: QueryParams) {
```

Comment:
When I'm thinking about it, maybe `stringifyUrl` shouldn't be in this module even. If I wrote that code, I would try to extract it to the separate module to clearly see the dependencies and make it work with as few dependencies as possible so that it wouldn't know too much about the app. Now it's not the case, and because it's in the fetcher module, it's not as apparent as it could be in the separate one.